src/main/webapp/app/service/trial.service.ts

```diff
@@ -342,4 +362,20 @@ export class TrialService {
         }
         return result;
     }
+    getErrorResponse(error: HttpErrorResponse, type: string) {
+        if (error.status === 400) {
+            alert('Sorry, your query is invalid.');
+        } else if (error.status === 404) {
+            alert('Sorry, we cannot find ' + type + '.');
+        } else if (error.status === 503) {
+            alert('Sorry, required data source is unavailable now.');
+        } else {
+            alert('Sorry, unexpected error happens. Our development team has been notified.');
```

Comment:
The message needs to be used in the callback of sendEmail, or at least after sending email.